Remotion/Data/DomainObjects.UnitTests/Mapping/TypeDefinitionChecker.cs

```diff
@@ -92,50 +61,63 @@ public void Check (ClassDefinition expectedDefinition, ClassDefinition actualDef
 
       foreach (var expectedDefinition in expectedDefinitions)
       {
-        var actualDefinition = actualDefinitions[expectedDefinition.ClassType];
-        Check(expectedDefinition, actualDefinition);
-        CheckDerivedClasses(expectedDefinition, actualDefinition);
+        var actualDefinition = actualDefinitions[expectedDefinition.Type];
+        if (expectedDefinition is ClassDefinition expectedClassDefinition)
+        {
+          if (actualDefinition is not ClassDefinition actualClassDefinition)
+          {
+            Assert.Fail("The two type definitions must have a matching type.");
+            return;
```

Comment:
```suggestion
```
Since Assert.Fail throws, a return isn't needed.

Reply:
It is needed here because `actualClassDefinition` is not defined in the else part